script.js

```diff
@@ -1,149 +1,638 @@
+
+function navAnimate(dir1, dir2) {
+  navlinksArr.forEach((nav, i) => {
+    nav.classList.replace(`slide-${dir1}-${i + 1}`, `slide-${dir2}-${i + 1}`);
+  });
+}
+
+function toggleMenu() {
+  menuBar.classList.toggle('change');
+
+  menuOverlay.classList.toggle('menu-overlay-active');
+
+  if (menuOverlay.classList.contains('menu-overlay-active')) {
+    menuOverlay.classList.replace('overlay-slide-left', 'overlay-slide-right');
+
+    navAnimate('out', 'in');
+  } else {
+    menuOverlay.classList.replace('overlay-slide-right', 'overlay-slide-left');
+    navAnimate('in', 'out');
+    cards.forEach(card => {
+      card.cardEl.classList.remove('hide');
     });
   }
+}
+menuBar.addEventListener('click', toggleMenu);
+/* --nav animation ends here------------------ */
+const navInput = document.querySelector('.menu__input-value');
+navInput.addEventListener('input', function (e) {
+  e.preventDefault();
+  const value = e.target.value.toLowerCase();
+  cards.forEach(card => {
+    const isVisible = card.title.toLowerCase().includes(value);
+    card.cardEl.classList.toggle('hide', !isVisible);
+  });
+});
+// console.log(cards);
 
-  header.insertAdjacentHTML('beforeend', headerMarkup);
-  const aboutMovie = document.getElementById('about-movie');
-  const moviePhoto = document.querySelector('.movie-photo');
+const documentary = document.querySelector('.documentary');
+documentary.addEventListener('click', function () {
+  cards.forEach(card => {
+    const isVisible = card.genre.toLowerCase().includes('documentary');
+
+    card.cardEl.classList.toggle('hide', !isVisible);
+  });
+});
 
-  cardElement.addEventListener('mouseover', function (e) {
-    aboutMovie.classList.remove('hidden');
-    moviePhoto.style.backgroundImage = `url(${movie.poster_url})`;
+const action = document.querySelector('.action');
+
+action.addEventListener('click', function () {
+  cards.forEach(card => {
+    const isVisible = card.genre.toLowerCase().includes('action');
+
+    card.cardEl.classList.toggle('hide', !isVisible);
   });
+});
 
-  cardElement.addEventListener('mouseleave', function () {
-    aboutMovie.classList.add('hidden');
+const recently = document.querySelector('.recently');
+// cards.forEachrecently(card => {
+//   console.log(card.year);
+// });
+recently.addEventListener('click', function () {
+  cards.forEach(card => {
+    if (card.year > 2011) {
+      const isVisible = card.year;
+      card.cardEl.classList.toggle('hide', !isVisible);
+    }
   });
+});
+/* FOOTER */
+
+const timer = document.querySelector('.timer');
+
+function logOutTimer() {
+  let time = 100;
+
+  const tick = setInterval(function () {
+    const min = Math.trunc(time / 60)
+      .toString()
+      .padStart(2, '0');
+    const sec = String(Math.floor(time % 60)).padStart(2, '0');
+
+    if (time === 0) {
+      clearInterval(tick);
+    }
+    time--;
+
+    timer.textContent = `You will logout in ${min}:${sec}`;
+  }, 1000);
```

Comment:
Nice implementation. The exercise states that you should be able to see how long you have been on this site. How can you easily change this function to enable such a functionality?

Additionally, for the untrained eye it may be a little unclear what all of this code does. can we move some of the code into a separate function used to format a time into a human readable text?